Sources/Foundation/Decimal.swift

```diff
@@ -1687,64 +1687,74 @@ public func NSDecimalString(_ dcm: UnsafePointer<Decimal>, _ locale: Any?) -> St
         return "0"
     }
 
+    var decimalSeparator: String!
+    if let locale = locale as? Locale {
+        decimalSeparator = locale.decimalSeparator
+    } else if let dictionary = locale as? [NSLocale.Key: String] {
+        decimalSeparator = dictionary[NSLocale.Key.decimalSeparator]
+    } else if let dictionary = locale as? [String: String] {
+        decimalSeparator = dictionary[NSLocale.Key.decimalSeparator.rawValue]
+    } else {
+        decimalSeparator = "."
+    }
+    let decimalSeparatorCChar = decimalSeparator.utf8CString // should be accessed with withUnsafeBufferPointer
```

Comment:
Wont this cause an extra memory allocation to create the `utf8CString`, could  a `UTF8View` be used and  later  iterated byte by byte rather than using the `memcpy` below?